[dialog-sx-since-last-run] Fix handling of last run dialog on file open

nomis opened the pull request:
When the "since last run" notification dialog is disabled on file open, it leaks a reference to GncSxInstanceModel and any creation errors won't be displayed. Avoid returning from the function too early.

jralls reviewed (changes requested):
Also unref `inst_model` right after `gnc_sx_instance_model_effect_change` and to free `auto_created_txns` as the first statement in the `else`.

`gnc_ui_sx_since_last_run_dialog` leaks the created_txns list if the user cancels, could you fix that too by moving the free and null out of the `GTK_RESPONSE_OK` block to after the `switch` statement?

nomis commented:
> Also unref `inst_model` right after `gnc_sx_instance_model_effect_change`

I can't do that because it's used by the call to `gnc_ui_sx_since_last_run_dialog()`.

> `gnc_ui_sx_since_last_run_dialog` leaks the `created_txns` list if the user cancels, could you fix that too by moving the free and null out of the `GTK_RESPONSE_OK` block to after the switch statement?

It's not freed in the `GTK_RESPONSE_OK` block of `dialog_response_cb`?

jralls commented:
> I can't do that because it's used by the call to gnc_ui_sx_since_last_run_dialog().

Ah, so it is. 

> It's not freed in the GTK_RESPONSE_OK block of dialog_response_cb?

That is where it's freed, and that's the problem: If the user clicks Cancel instead of OK that block doesn't execute and the list is leaked.

nomis commented:
> > It's not freed in the GTK_RESPONSE_OK block of dialog_response_cb?
> 
> That is where it's freed, and that's the problem: If the user clicks Cancel instead of OK that block doesn't execute and the list is leaked.

Ok... the indentation in that switch block is wrong. I've fixed that and moved the free to where `gnc_close_gui_component()` is called.

jralls commented:
Thanks!
